update release_dev.txt with recently merged PRs (#5557)
scikit-image/scikit-image: doc/release/release_dev.rst

https://scikit-image.org

A highlight of this release is the addition of the popular scale-invariant
feature transform (SIFT) feature detector and descriptor. This release also

> introduces a perceptual blur metric, new pixel graph algorithms and most

Reviewer: ```suggestion
introduces a perceptual blur metric, new pixel graph algorithms, and most
```

  were added to ``skimage.color.lab2rgb``, ``skimage.color.rgb2lab``,
  ``skimage.color.xyz2lab``, ``skimage.color.lab2xyz``,
  ``skimage.color.xyz2luv`` and ``skimage.color.luv2xyz``.
- ``skimage.filters.threshold_multiotsu`` has a new ``hist`` keyword argument
  to allow use with a user-supplied histogram. (gh-5543)
- ``skimage.restoration.denoise_bilateral`` added support for images containing
  negative values. (gh-5527)
- The ``skimage.feature`` functions ``blob_dog``, ``blob_doh`` and ``blob_log``
  now support a ``threshold_rel`` keyword argument that can be used to specify
  a relative threshold (in range [0, 1]) rather than an absolute one. (gh-5517)
- Implement lazy submodule importing (gh-5101)

> - Weighted estimation of geometric transform matrices (gh-5601)

Reviewer: ```suggestion
- Implement weighted estimation of geometric transform matrices (gh-5601)
```

- Update reference link documentation in the ``adjust_sigmoid`` function.
- Fix reference to multiscale_basic_features in TrainableSegmenter.
- Slight ``shape_index`` docstring modification to specify 2D array.
- Add stitching gallery example (gh-5365)
- Add draft SKIP3: transition to scikit-image 1.0 (gh-5475)
- Mention commit messages in the contribution guidelines. (gh-5504)

> - Fix and standardize docstrings for blob detection functions (gh-5547)
> - Updated the User Guide to reflect usage of ``channel_axis`` rather than

Reviewer: ```suggestion
- Fix and standardize docstrings for blob detection functions. (gh-5547)
- Update the User Guide to reflect usage of ``channel_axis`` rather than
```

  ``multichannel``. (gh-5554)

> - Update the user guide to use channel_axis rather than multichannel (gh-5556)

Reviewer: Merge with the above?

- Add hyperlinks to referenced documentation places. (gh-5560)

> - Update branching instructions to change the location of the pooch repo

Reviewer: ```suggestion
- Update branching instructions to change the location of the pooch repo.
```

  (gh-5565)

> - add Notes and References section to the Cascade class docstring (gh-5568)

Reviewer: ```suggestion
- Add Notes and References section to the Cascade class docstring. (gh-5568)
```

> - Clarify 2D vs nD in skimage.feature.corner docstrings (gh-5569)

Reviewer: Not sure, because now there's a sequence of items ending with no period (just like a standard commit message)... :confused:

> - Fixed math formulas in plot_swirl.py example (gh-5574)

Reviewer: ```suggestion
- Fix math formulas in plot_swirl.py example. (gh-5574)
```

  ``skimage.restoration.denoise_bilateral`` now release the GIL, enabling
  multithreaded use.
- A ``skimage.color.label2rgb`` performance regression was addressed.

> - Improved numerical precision in ``CircleModel.estimate``. (gh-5190)
> - Added default keyword argument values to

Reviewer: ```suggestion
- Improve numerical precision in ``CircleModel.estimate``. (gh-5190)
- Add default keyword argument values to
```

  can be used to override this behavior when desired.
- Color conversion functions now have a new ``channel_axis`` keyword argument
  (see **New Features** section).
- SLIC superpixel segmentation outputs may differ from previous versions for
  data that was not already scaled to [0, 1] range. There is now an automatic
  internal rescaling of the input to [0, 1] so that the ``compactness``
  parameter has an effect that is independent of the input image's scaling.

> - A bug fix to the phase nomalization applied within

Reviewer: ```suggestion
- A bug fix to the phase normalization applied within
```

- Fixed NaN issue in ``skimage.filters.threshold_otsu``.
- Fix ``skimage.feature.blob_dog`` docstring example and normalization.
- Fix uint8 overflow in ``skimage.exposure.adjust_gamma``.
- Work with pooch 1.5.0 for fetching data (gh-5529).
- The ``offsets`` attribute of ``skimage.graph.MCP`` is now public. (gh-5547)
- Fix io.imread behavior with pathlib.Path inputs (gh-5543)
- Make scikit-image imports from Pooch, compatible with pooch >= 1.5.0.
  (gh-5529)
- Fix several broken doctests and restore doctesting on GitHub Actions.
  (gh-5505)
- Fix broken doctests in ``skimage.exposure.histogram`` and
  ``skimage.measure.regionprops_table``. (gh-5522)
- Rescale image consistently during SLIC superpixel segmentation. (gh-5518)

> - Corrected phase correlation in ``skimage.register.phase_cross_correlation``.

Reviewer: ```suggestion
- Correct phase correlation in ``skimage.register.phase_cross_correlation``.
```

- Ensure that README.txt has write permissions for subsequent imports.
- Decorators for helping with the transition between the keyword argument
  multichannel and channel_axis.

> - Missing import in lch2lab docstring example (gh-5998)

Reviewer: ```suggestion
- Add missing import in lch2lab docstring example (gh-5998)
```

- Simplify resize implementation using new SciPy 1.6 zoom option.
- Fix duplicate test function names in ``test_unsharp_mask.py``.
- Benchmarks: ``fix ResizeLocalMeanSuite.time_resize_local_mean`` signature.

> - prefer use of new-style NumPy random API in tests (gh-5450)

Reviewer: ```suggestion
- Prefer use of new-style NumPy random API in tests (gh-5450)
```

- Add fixture enforcing SimpleITK I/O in test_simpleitk.py (gh-5526)
- MNT: Remove unused stat import from skimage data (gh-5566)
- MAINT: Remove unused imports (gh-5595)
- MAINT: Refactor duplicated tests, remove unnecessary assignments and
  variables (gh-5596)
- Remove obsolete lazy import (gh-5992)
- Lazily load data_dir into the top-level namespace (gh-5996)
- Update scipy requirement to 1.4.1 and use scipy.fft instead of scipy.fftpack
  (gh-5999)

> - removed lines generating Requires metadata (gh-6017)

Reviewer: ```suggestion
- Remove lines generating Requires metadata (gh-6017)
```

> - update pyproject.toml to handle Python 3.10 and Apple arm64 (gh-6022)

Reviewer: ```suggestion
- Update pyproject.toml to handle Python 3.10 and Apple arm64 (gh-6022)
```
